packages/evolution-common/src/services/baseObjects/Address.ts

```diff
@@ -21,43 +26,44 @@ export const addressAttributes = [
     'streetName',
     'streetNameHomogenized',
     'streetNameId',
-    'streetNameInternalId',
     'municipalityName',
     'municipalityCode',
     'postalMunicipalityName',
     'region',
     'country',
     'postalCode',
-    'addressId',
-    'internalId',
-];
-
-/** homogenized street name:
- * - replace latin characters with their non-latin equivalent
- * - lowercase
- * - no dash
- * - includes non-abbreviated street type/prefix and/or suffix (e.g. "rue", "avenue", "boulevard", etc.)
- * - includes non-abbreviated orientation
- * - non-abbreviated saint/sainte
- * - trimed and trimed start
- * example: rue de la gauchetiere ouest | 5e avenue nord | rang du petit saint jean | 30e rue
+    'addressId'
+] as const;
+
+/**
+ * Note about ? and Optional type:
+ * We keep the ? so it accepts empty params
+ * We add the Optional type to make it obvious that the attribute is optional everywhere it appears
  */
 export type AddressAttributes = {
     civicNumber: number;
-    civicNumberSuffix?: string;
-    unitNumber?: number;
-    streetName: string;
-    streetNameHomogenized?: string;
-    streetNameId?: string;
-    streetNameInternalId?: string;
+    civicNumberSuffix?: Optional<string>; // example: A, B, C
+    unitNumber?: Optional<string>; // example: 101, 202
+    streetName: string; // no abbreviation, with latin characters and capital letters
+    /** homogenized street name:
+     * - replace latin characters with their non-latin equivalent
+     * - lowercase
+     * - no dash
+     * - includes non-abbreviated street type/prefix and/or suffix (e.g. "rue", "avenue", "boulevard", etc.)
+     * - includes non-abbreviated orientation
+     * - non-abbreviated saint/sainte
+     * - trimed and trimed start
+     * example: 30th street | saint john boulevard | rue de la gauchetiere ouest | 5e avenue nord | rang du petit saint jean | 30e rue
+     */
+    streetNameHomogenized?: Optional<string>; // should be unique by municipality
+    streetNameId?: Optional<string>; // official street name id
     municipalityName: string;
-    municipalityCode?: string;
-    postalMunicipalityName?: string;
+    municipalityCode?: Optional<string>; // official code for the municipality
+    postalMunicipalityName?: Optional<string>; // some municipalities have a different name for postal addresses
     region: string;
     country: string;
-    postalCode?: string;
-    addressId?: string;
-    internalId?: string;
+    postalCode?: Optional<string>;
+    addressId?: Optional<string>; // official address id
```

Comment:
Il ne faut pas être avare de commentaires... official comme quoi? ça sort d,où? Que vont comprendre ceux qui vont développer l'enquête OD de l'Antarctique?

Reply:
added a todo for international and info to address id